server/src/test/java/keywhiz/service/permission/AlwaysAllowDelegatingPermissionCheckTest.java

```diff
@@ -0,0 +1,73 @@
+package keywhiz.service.permission;
+
+import com.codahale.metrics.Counter;
+import com.codahale.metrics.MetricRegistry;
+import java.util.Objects;
+import keywhiz.service.permissions.Action;
+import keywhiz.service.permissions.AlwaysAllowDelegatingPermissionCheck;
+import keywhiz.service.permissions.AlwaysFailPermissionCheck;
+import keywhiz.service.permissions.KeywhizPrincipal;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.when;
+
+public class AlwaysAllowDelegatingPermissionCheckTest {
+
+  @Rule public MockitoRule mockito = MockitoJUnit.rule();
+
+  @Mock AlwaysFailPermissionCheck alwaysFail;
+
+  MetricRegistry metricRegistry;
+  AlwaysAllowDelegatingPermissionCheck alwaysAllow;
+  KeywhizPrincipal principal;
+  Objects target;
+
+  @Before
+  public void setUp() {
+    metricRegistry = new MetricRegistry();
+    metricRegistry.register("permissionCheckCount", new Counter());
+    metricRegistry.register("permissionCheckDeniedCount", new Counter());
+    alwaysAllow = new AlwaysAllowDelegatingPermissionCheck(metricRegistry, alwaysFail);
+  }
+
+  @Test public void delegateIsAllowedReturnsTrue() {
+    when(alwaysFail.isAllowed(principal, Action.ADD, target)).thenReturn(true);
+
+    boolean permitted = alwaysAllow.isAllowed(principal, Action.ADD, target);
+
+    assertThat(permitted).isEqualTo(true);
+    assertThat(metricRegistry.counter("permissionCheckCount").getCount()).isEqualTo(1);
+    assertThat(metricRegistry.counter("permissionCheckDeniedCount").getCount()).isEqualTo(0);
+  }
+
+  @Test public void delegateIsAllowedReturnsFalse() {
+    when(alwaysFail.isAllowed(principal, Action.ADD, target)).thenReturn(false);
```

Comment:
Mockito includes an `any()` matcher which is a nice way of communicating to the reader "I don't care about the actual values of the parameters". For cases like this where the desired behavior is to return false no matter what it can make the intent more clear.